Utilities/sysinfo.cpp

```diff
@@ -51,6 +50,22 @@ bool utils::has_xop()
 	return g_value;
 }
 
+int utils::is_haswell_or_broadwell()
+{
+	static const int g_model = ((get_cpuid(1, 0)[0] >> 4) & 0xf) + (((get_cpuid(1, 0)[0] >> 16) & 0xf) << 4);
+	//static const bool g_value = g_model == 0x3c || g_model == 0x3f || g_model == 0x45 || g_model == 0x46 || // Haswell
```

Comment:
you could modify this to use an int with a ternary expression

Reply:
that would take up an unnecessary amount of space and hurt readability considering how many conditions there are being checked

Reply:
that would take up an unnecessary amount of space and hurt readability considering how many conditions there are being checked